browser/src/helpers/blobs.tsx

```diff
@@ -1,9 +1,70 @@
-import React from "react";
-import { Pattern, Prompt } from "../types/common/server-api";
-
-export function renderBlob(
-  pattern: Pattern,
-  prompt: Prompt
-): React.ReactElement {
-  return <></>;
+import React, {useCallback, useEffect, useMemo, useState} from "react";
+import {Contribution, Prompt} from "../types/common/server-api";
+import { sha256 } from "ethers/lib/utils";
+import Blob from "../components/Blob";
+import { OrbitControls } from "@react-three/drei";
+import { Canvas } from "@react-three/fiber";
+import {Placeholder} from "../components/ContributionSection";
+
+function toHex(str: string) {
+  return str
+    .split("")
+    .map((_, i) => str.charCodeAt(i).toString(16))
+    .join("");
+}
+
+function fetch(message: string, idx: number, val_start: number, range: number) {
+  const sub = message.substring(idx * 8, idx * 8 + 8);
+  const eightCount = sub.split("8").length;
+  const float = parseInt(sub, 16) / 0xffffffff;
+  return float * range * eightCount + val_start;
+}
+
+export const PromptColours: Record<Prompt, number> = {
+    [Prompt.LooksLike]: 0,
+    [Prompt.WeNeed]: 0.25,
+    [Prompt.Example]: 0.5,
+};
+
+export function renderBlob(contribution: Contribution): React.ReactElement {
+  // 64 chars long sha256 str
+  const contrib = `${contribution.author.walletId}@${contribution.pattern}: ${contribution.response}`;
+
+  const [message, setMessage] = useState(sha256(`0x${toHex(contrib)}`));
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setMessage(sha256(`0x${toHex(contrib)}`));
+    }, 500);
+
+    return () => {
+      // teardown
+      clearTimeout(handler);
+    };
+  }, [contrib]);
+
+  return (
+    <Canvas camera={{ position: [0, 0, 20], fov: 50 }}>
```

Comment:
I wonder how many threejs/webgl instances can we support rendering at a time w/o hitting perf issues, some other sites will just use a single canvas in the page and then move the camera on scroll/swap the objects but I think this is more straightforward